### PR TITLE
Small clean up around KryoAtomicCoder

### LINKED ISSUE #1172: Clean up Kryo stuff
`KryoAtomicCoder` has blown up quite a bit with the addition of several custom serializers and could use a bit of clean up. 

- [ ] Move `KSerializer` classes into a subpackage, maybe `coders.serializers`
- [ ] Add a `ScioRegistrar` for our custom serializers and load in `KryoAtomicCoder` to get rid of the `k.forSubclass` bloat.

### PR DESCRIPTION
**Pull request by regadas**
Closes #1172 #1220

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -95,6 +95,8 @@ val commonSettings = Sonatype.sonatypeSettings ++ assemblySettings ++ Seq(
 
   // protobuf-lite is an older subset of protobuf-java and causes issues
   excludeDependencies += "com.google.protobuf" % "protobuf-lite",
+  // hamcrest-core overlaps with hamcrest-all and causes issues in IntelliJ
+  excludeDependencies += "org.hamcrest" % "hamcrest-core",
```
Comment (nevillelyh) [inline build.sbt]: Is this included due to stale branch? Rebase on master?

Comment (regadas) [reply build.sbt]: I rebased on master

Comment (regadas) [reply build.sbt]: bad rebase apparently.

**Comment by nevillelyh** [comment]
LGTM but wonder how much this will interfere with @jto's ser/de work?

**Comment by regadas** [comment]
@nevillelyh don't actually know how much will affect

**Comment by regadas** [comment]
From a quick look at @jto branch I think it's ok to merge this now.

**Pull request merged**